op-node/sources/eth_client.go

```diff
@@ -307,6 +307,14 @@ func (s *EthClient) InfoByLabel(ctx context.Context, label eth.BlockLabel) (eth.
 	return s.headerCall(ctx, "eth_getBlockByNumber", label)
 }
 
+func (s *EthClient) BSCInfoByLabel(ctx context.Context, label eth.BlockLabel) (eth.BlockInfo, error) {
+	// can't hit the cache when querying the head due to reorgs / changes.
+	if label == eth.Finalized {
+		return s.headerCall(ctx, "eth_getFinalizedBlock", numberID(15))
```

Comment:
What about downgrading to `s.headerCall(ctx, "eth_getBlockByNumber", label)` when `eth_getFinalizedBlock` responses `methodNotFoundError`?

So that it can cover both Ethereum and BSC nodes.